### PR TITLE
Move App to `src` redux

### PR DESCRIPTION
**Pull request by Hypolite**
Replace #3413 after a bad rebase.

Up-to-date and working.

### CODE REVIEW

File: boot.php [hunk 1]
```diff
@@ -2425,27 +1413,27 @@ function clear_cache($basepath = "", $path = "") {
...
 	}
 
-	if (is_writable($path)){
+	if (is_writable($path)) {
 		if ($dh = opendir($path)) {
 			while (($file = readdir($dh)) !== false) {
-				$fullpath = $path."/".$file;
-				if ((filetype($fullpath) == "dir") and ($file != ".") and ($file != "..")) {
+				$fullpath = $path . "/" . $file;
+				if ((filetype($fullpath) == "dir") and ( $file != ".") and ( $file != "..")) {
```
Comment (annando) [inline boot.php]: I guess these spaces were applied by accident?

File: boot.php [hunk 2]
```diff
@@ -2669,12 +1657,12 @@ function infinite_scroll_data($module) {
 
 		// try to get the uri from which we load the content
 		foreach ($_GET AS $param => $value) {
-			if (($param != "page") AND ($param != "q")) {
+			if (($param != "page") AND ( $param != "q")) {
```
Comment (annando) [inline boot.php]: Here as well

File: util/typo.php [hunk 3]
```diff
@@ -1,59 +1,67 @@
...
-	$files = glob('object/*.php');
-	foreach($files as $file) {
-        passthru("$phpath -l $file", $ret); $ret===0 or die();
-	}
+echo "Directory: object\n";
+$files = glob('object/*.php');
+foreach ($files as $file) {
+	passthru("$phpath -l $file", $ret);
+	$ret === 0 or die();
+}
```
Comment (annando) [inline util/typo.php]: Shouldn't the "src" now be checked as well?

File: boot.php (hunk 1 above)
Comment (Hypolite) [inline boot.php]: Indeed, nice catch!

File: util/typo.php (hunk 3 above)
Comment (Hypolite) [inline util/typo.php]: It's a very good point, thanks!

**Comment by annando** [comment]
Since this is some huge bunch of code lines, I would like to see a second pair of eyes having reviewed this,

**Comment by Hypolite** [comment]
I'm sorry I'm going against my own principles, it looks like I'm doing as if #3381 doesn't exist since we haven't heard about @Quix0r for more than 10 days now.

**Comment by tobiasd** [comment]
Looks good for me. The only thing was in mod/help.php the added HTML code, wouldn't it be better to wrap a new template around it?

**Comment by Hypolite** [comment]
It was just a fix for `code` in titles, I don't think it warrants more changes.

**Comment by tobiasd** [comment]
With this PR pulled in, my node stopped working. git bisect boiled down to the commit "Add basepath to App" as the first *bad* one (regardless of PHP5.6 or 7.0) on Debian/Raspbial Jessie.

In the Vagrant VM (Ubuntu Trusty with PHP5.5.9) it works.

File: util/typo.php (hunk 3 above)
Comment (Quix0r) [inline util/typo.php]: Has this been fixed? Seems to be still open, according to this WUI.

Comment (Hypolite) [inline util/typo.php]: Yes it is. This is open because I didn't change the line @annando added his comment on.

**Pull request merged**